processor/src/main/java/org/mapstruct/ap/internal/model/NestedTargetPropertyMappingHolder.java

```diff
@@ -218,5 +508,123 @@ public NestedTargetPropertyMappingHolder build() {
             }
             return result;
         }
+
+        private PropertyMapping createPropertyMappingForNestedTarget(MappingOptions mappingOptions,
+            PropertyEntry targetProperty, SourceReference sourceReference, boolean forceUpdateMethod) {
+            PropertyMapping propertyMapping = new PropertyMapping.PropertyMappingBuilder()
+                .mappingContext( mappingContext )
+                .sourceMethod( method )
+                .targetProperty( targetProperty )
+                .targetPropertyName( targetProperty.getName() )
+                .sourceReference( sourceReference )
+                .existingVariableNames( existingVariableNames )
+                .dependsOn( mappingOptions.collectNestedDependsOn() )
+                .forgeMethodWithMappingOptions( mappingOptions )
+                .forceUpdateMethod( forceUpdateMethod )
+                .forgedNamedBased( false )
+                .build();
+            return propertyMapping;
+        }
+
+        /**
+         * If a single target mapping has a valid {@link SourceReference} and the {@link SourceReference} has more
+         * then 0 {@link PropertyEntry} and if the {@code map} does not contain an entry with the extracted key then
+         * an entry with the extracted key and an empty list is added.
+         *
+         * @param map that needs to be populated
+         * @param singleTargetReferences to use
+         * @param keyExtractor to be used to extract a key
+         */
+        private <K> void populateWithSingleTargetReferences(Map<K, List<Mapping>> map,
+            List<Mapping> singleTargetReferences, Extractor<SourceReference, K> keyExtractor) {
+            if ( singleTargetReferences != null ) {
+                //This are non nested target references only their property needs to be added as they most probably
+                // define it
+                for ( Mapping mapping : singleTargetReferences ) {
+                    if ( mapping.getSourceReference() != null && mapping.getSourceReference().isValid()
+                        && !mapping.getSourceReference().getPropertyEntries().isEmpty() ) {
+                        //TODO is this OK? Why there are no propertyEntries? For the Inverse LetterMapper for example
```

Comment:
@gunnarmorling, @sjaakd Did you had a look at this one as well? Is this an issue or not? Does it make sense that the source references property entries are empty?